kratos/utilities/auxiliar_model_part_utilities.cpp

```diff
@@ -0,0 +1,303 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "utilities/auxiliar_model_part_utilities.h"
+
+namespace Kratos
+{
+void AuxiliarModelPartUtilities::RemoveElementAndBelongings(
+    IndexType ElementId, Flags IdentifierFlag, IndexType ThisIndex)
+{
+    auto& r_array_nodes = mrModelPart.Nodes(ThisIndex);
+    #pragma omp parallel for
+    for(int i=0; i<static_cast<int>(r_array_nodes.size()); ++i) {
+        auto it_node = r_array_nodes.begin() + i;
+        it_node->Set(IdentifierFlag, true);
+    }
+
+    // TODO: Add OMP
+    for (auto& elem : mrModelPart.Elements(ThisIndex)) {
+        if (elem.Id() != ElementId) {
+            for (auto& node : elem.GetGeometry()) {
+                node.Set(IdentifierFlag, false);
+            }
+        }
+    }
+    bool condition_to_remove;
+    for (auto& cond : mrModelPart.Conditions(ThisIndex)) {
+        condition_to_remove = true;
+        for (auto& node : cond.GetGeometry()) {
+            if (node.IsNot(IdentifierFlag)) {
+                condition_to_remove = false;
+                break;
+            }
+            if (condition_to_remove) cond.Set(IdentifierFlag);
+        }
+    }
+
+    mrModelPart.RemoveElement(ElementId, ThisIndex);
+    mrModelPart.RemoveConditions(IdentifierFlag);
+
+    // TODO: Add OMP
+    for(std::size_t i=0; i<r_array_nodes.size(); ++i) {
+        auto it_node = r_array_nodes.begin() + i;
+        if (it_node->Is(IdentifierFlag)) {
+            mrModelPart.RemoveNode(it_node->Id(), ThisIndex);
```

Comment:
doing it one by one it is very very slow. You should mark them TO_ERASE and then remove them at once.